### ISSUE TITLE
Postgres.app should not crash when launched from the wrong directory

### ISSUE THREAD

**Issue by jakob** [PostgresApp/PostgresApp#639]
All the dylibs in Postgres.app use absolute paths. This makes it easier to build things (eg. ruby gems or python modules) that link against our libpq.

Of course, the downside is that Postgres.app has to be located exactly in `/Applications/Postgres.app`.

A long time ago, Postgres.app warned users when they tried to open it from a wrong location. However, ever since version 2.0 Postgres.app itself links with libpq, and therefore it crashes on launch if it's in the wrong location.

A possible workaround would be instead of linking with libpq at compile time, would be to use dlopen() / dlsym() to load the required libraries at runtime, after we checked to make sure it is located in the correct location. Then we could show a nice warning message explaining why Postgres.app uses absolute paths.

**Comment by jakob** [comment]
Related issues:
- #469
- #562 
- #601
- https://github.com/PostgresApp/PostgresApp/issues/638#issuecomment-955606119

**Comment by jakob** [comment]
An even simpler method would be to add a static libpq for Postgres.app itself.

**Comment by jakob** [comment]
I've now changed Postgres.app so that it links with a static libpq. So now it no longer crashes when you launch it from somewhere else than /Applications

However, it still fails the gatekeeper check if you don't move it to /Applications. I hoped it wouldn't check everything in the bundle, but it seems that it does...

So you would need to move it to /Applications and launch it from there at least once, then you can move it, and get an error message telling you to put it back. Not a big improvement....

**Comment by jakob** [comment]
You can try the builds from this release to see for yourself:
https://github.com/PostgresApp/PostgresApp/releases/tag/v2.5.3

**Comment by jakob** [comment]
I've made some progress! I replaced all the absolute launch paths with relative paths using `@loader_path`. The result is an app that passes the gatekeeper check and can be launched directly from the disk image!

First test build here: https://github.com/PostgresApp/PostgresApp/releases/tag/v2.5.4

To Do:
- [ ] Check if this also works with older versions of PostgreSQL
- [ ] Check if this works on all supported versions of macOS
- [ ] Check if compiling `pg` gem or `psycopg2` works with this build
- [ ] Check if compiling PostgreSQL extesnions works with this build
